src/dosinst.h

```diff
@@ -498,7 +498,7 @@ remove_tail(char *path)
 }
 
 
-char	installdir[BUFSIZE];	/* top of the installation dir, where the
+char	installdir[MAX_PATH-9];	/* top of the installation dir, where the
```

Comment:
You changed the size here, but you didn't change the size in line 515 where this is used.
Buffer overflow may occur.
BTW, why -9?